harness/determined/pytorch/deepspeed/_mpu.py

```diff
@@ -0,0 +1,50 @@
+from dataclasses import dataclass
+
+from deepspeed.runtime.pipe import topology
+
+from determined import _core
+
+
+@dataclass
+class ModelParallelUnit:
+    """
+    This class contains the functions we expect in order to accurately carry out parallel training.
+    For custom model parallel training, you need to subclass and override the functions before
+    passing it to the :class:`~determined.pytorch.deepspeed.DeepSpeedTrialContext` by calling
+    ``context.wrap_mpu(mpu)``.
+    """
+
+    data_parallel_rank: int
+    data_parallel_world_size: int
+    # Whether the rank should return metrics from training and evaluation steps.
+    # This is usually true for ranks in the last stage of pipeline parallel or
+    # model parallel training.
+    should_report_metrics: bool = True
```

Comment:
I'm thinking that `should_report_metrics` and `should_build_data_loader` should not have defaults.

If the users is advanced enough to need this struct, they can handle two extra bools.  If you don't need to set values for the bools, then you probably don't need the struct at all, right?